eng/scripts/Publish-MavenPackages.ps1

```diff
@@ -159,12 +178,33 @@ foreach ($packageDetail in $packageDetails) {
       Write-Information "Skipping release of staging repository because stage only is set to false."
     }
     else {
-      Write-Information "Releasing staging repostiory $stagedRepositoryId"
-      Write-Information "mvn org.sonatype.plugins:nexus-staging-maven-plugin:rc-release `"-DstagingRepositoryId=$stagedRepositoryId`" `"-DnexusUrl=https://oss.sonatype.org`" `"-DrepositoryId=target-repo`" `"-DserverId=target-repo`" `"-Drepo.username=$RepositoryUsername`" `"-Drepo.password=`"`"$RepositoryPassword`"`"`" `"--settings=$PSScriptRoot\..\maven.publish.settings.xml`""
-      mvn org.sonatype.plugins:nexus-staging-maven-plugin:rc-release "-DstagingRepositoryId=$stagedRepositoryId" "-DnexusUrl=https://oss.sonatype.org" "-DrepositoryId=target-repo" "-DserverId=target-repo" "-Drepo.username=$RepositoryUsername" "-Drepo.password=""$RepositoryPassword""" "--settings=$PSScriptRoot\..\maven.publish.settings.xml"
+      $attempt = 0
+      while($attempt++ -lt 3)
+      {
+        Write-Information "Releasing staging repostiory $stagedRepositoryId, attempt $attempt"
+        Write-Information "mvn org.sonatype.plugins:nexus-staging-maven-plugin:rc-release `"-DstagingRepositoryId=$stagedRepositoryId`" `"-DnexusUrl=https://oss.sonatype.org`" `"-DrepositoryId=target-repo`" `"-DserverId=target-repo`" `"-Drepo.username=$RepositoryUsername`" `"-Drepo.password=`"`"[redacted]`"`"`" `"--settings=$PSScriptRoot\..\maven.publish.settings.xml`""
+        mvn org.sonatype.plugins:nexus-staging-maven-plugin:rc-release "-DstagingRepositoryId=$stagedRepositoryId" "-DnexusUrl=https://oss.sonatype.org" "-DrepositoryId=target-repo" "-DserverId=target-repo" "-Drepo.username=$RepositoryUsername" "-Drepo.password=""$RepositoryPassword""" "--settings=$PSScriptRoot\..\maven.publish.settings.xml"
+  
+        if($LASTEXITCODE -eq 0) {
+          "Package $($packageDetail.FullyQualifiedName) deployed"
```

Comment:
Did you want this to be Write-Host?

Reply:
I'm avoiding Write-Host because I want to pipe to file and compare runs.  I could use transcripts, but piping is easier.

Reply:
Is it that useful given we don't correctly write other context to the pipeline? Also there is nothing stopping use from redirecting write-host to a file as well.

Reply:
I'd like to get more info on redirecting write-host to a file.   From what I've seen, you can only redirect it from outside of powershell, i.e. Because powershell doesn't allow piping of Write-Host, you have to have an external shell redirect the output of pwsh to a file.